### PR TITLE
Guess the right name for COMPOSE_PROJECT_NAME

### PR DESCRIPTION
**Pull request by mstriemer**

### CODE REVIEW

File: docs/topics/install/docker.rst [hunk 1]
```diff
@@ -103,6 +103,13 @@ on each container. Here is an example of running the ``py.test`` command on the
 If you'd like to use a python debugger to interactively
 debug Django view code, check out the :ref:`debugging` section.
 
+.. note::
+    If you see an error like ``No such container: addonsserver_web_1`` please
+    ensure that your copy of addons-server is in a directory named
+    ``addons-server``. If you had an old checkout that was called ``olympia``
```
Comment (muffinresearch) [inline docs/topics/install/docker.rst]: Or you can set the env var

Comment (mstriemer) [inline docs/topics/install/docker.rst]: Totally, I'll add that.

**Comment by mstriemer** [comment]
I updated the `Makefile` to detect the name of the directory that you're in which is what `docker-compose` defaults to.

I also updated the documentation to reflect what to do if that's somehow not what the containers are called but hopefully this stops happening.

File: docs/index.rst [hunk 2]
```diff
@@ -15,5 +15,5 @@ Contents
    :maxdepth: 2
 
    topics/api/index
-   topics/install-olympia/index
+   topics/install/index
```
Comment (tofumatt) [inline docs/index.rst]: You'll want to update the README's link to install docs to make sure the README doesn't break.

**Comment by mstriemer** [comment]
This actually doesn't work since `docker-compose` strips (at least) `-`s.

File: docs/index.rst (hunk 2 above)
Comment (mstriemer) [inline docs/index.rst]: Thanks, updated.

**Comment by mstriemer** [comment]
This strips `-`s and `_`s now. Hopefully that's all people have in their folder names. If you have a space in the name then it won't work. I can't imagine someone doing that, but who knows.

File: Makefile [hunk 3]
```diff
@@ -2,7 +2,11 @@
 NUM_ADDONS=10
 NUM_THEMES=$(NUM_ADDONS)
 
-COMPOSE_PROJECT_NAME?=addonsserver
+# Get the name of the Makefile's directory for the docker container base name.
+mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+current_dir := $(notdir $(patsubst %/,%,$(dir $(mkfile_path))))
```
Comment (kumar303) [inline Makefile]: fail-kids

because Makefile

Comment (kumar303) [inline Makefile]: (not a comment suggesting you should change anything, just me trying to read the code is all)

Comment (mstriemer) [inline Makefile]: Definitely copy/pasted from http://stackoverflow.com/questions/18136918/how-to-get-current-directory-of-your-makefile.

Comment (kumar303) [inline Makefile]: but while I'm at it, it would be pretty nice to rewrite this as a fabric script or something

File: Makefile [hunk 4]
```diff
@@ -2,7 +2,11 @@
...
+
+COMPOSE_PROJECT_NAME?=$(shell echo "${current_dir}" | tr -d '-' | tr -d '_')
```
Comment (kumar303) [inline Makefile]: I hope this works all of the time

File: README.rst [hunk 5]
```diff
@@ -19,5 +19,5 @@ You can access the AMO stage environment at https://addons.allizom.org/
 You can join our mailing list at: https://mail.mozilla.org/listinfo/dev-addons
 
 .. _`addons.mozilla.org`: https://addons.mozilla.org
-.. _`install docs`: https://addons-server.readthedocs.org/en/latest/topics/install-olympia/docker.html
+.. _`install docs`: https://addons-server.readthedocs.org/en/latest/topics/install/docker.html
```
Comment (kumar303) [inline README.rst]: This should be a ref instead. Change the link to ``:ref:`install-with-docker```. This is a ref that's already defined. You can see the definition in `docker.rst` as `.. _install-with-docker:`.

**Comment by kumar303** [comment]
I don't think this is so necessary because new contributors will most likely always run `git clone ...` which by default will put the code in a directory called `addons-server`. But I suppose it wouldn't hurt to account for someone who clones to a custom directory.

r+wc

File: README.rst (hunk 5 above)
Comment (mstriemer) [inline README.rst]: This is in the README though so I don't think that will work? I don't think GitHub would know to link to https://addons-server.readthedocs.org so I don't know where the link would go.

Comment (kumar303) [inline README.rst]: whoops, ignore me. I mis-read the diff.

**Pull request merged**